### PR TITLE
[Sqlvm] `az sql vm enable-azure-ad-auth/validate-azure-ad-auth`: New commands for Sqlvm Azure AD authentication

### PR DESCRIPTION
**Pull request by qianliu-pdx**
**Related command**
az sql vm enable-azure-ad-auth
az sql vm validate-azure-ad-auth

**Description**
This PR is to add Azure CLI commands to enable and validate Azure AD authentication for SqlVM.
The Azure AD authentication is a new feature for SqlVM as described in the [public doc ](https://learn.microsoft.com/en-us/azure/azure-sql/virtual-machines/windows/configure-azure-ad-authentication-for-sql-vm?view=azuresql). 

Here is the [design specification of the Azure CLI work](https://microsoft-my.sharepoint.com/:w:/p/qianliu/ERVCuQ0YHd9Hp47Lit5mWpkBEVKf5lSjCaZCi5ShgYUeNA?e=PdLFqI) for the Azure AD authentication feature.

**Testing Guide**
The automatic test "test_sqlvm_aad_auth_negative" covers all different scenario of validation errors. The positive tests are tested manually because of the requirement of high Azure AD privileges from the account running the test. All the examples in the help page have been tested.

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Sqlvm

File: src/azure-cli/azure/cli/command_modules/sqlvm/_validators.py [hunk 1]
```diff
@@ -186,3 +190,322 @@ def validate_assessment_start_time_local(namespace):
...
+        :param namespace: The argparse namespace represents the arguments.
+        :type namespace: argpase.Namespace.
+    """
+
+    skip_client_validation = False
+    if hasattr(namespace, "skip_client_validation"):
+        skip_client_validation = getattr(namespace, "skip_client_validation")
+
+    if skip_client_validation is True:
+        print('Skipping client-side validation ...')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sqlvm/_validators.py:210]: If you want to warn customers, use `logger.warning()` instead of `print()`

Comment (qianliu-pdx) [reply src/azure-cli/azure/cli/command_modules/sqlvm/_validators.py]: fixed.

File: src/azure-cli/azure/cli/command_modules/sqlvm/custom.py [hunk 2]
```diff
@@ -392,6 +393,64 @@ def sqlvm_update(cmd, instance, sql_virtual_machine_name, resource_group_name, s
...
+    if sqlvm_object.server_configurations_management_settings is None:
+        sqlvm_object.server_configurations_management_settings = ServerConfigurationsManagementSettings()
+
+    sqlvm_object.server_configurations_management_settings.azure_ad_authentication_settings = AADAuthenticationSettings(client_id=msi_client_id if msi_client_id else '')
+
+    # Since it's a running operation, we will do the put and then the get to display the instance.
+    LongRunningOperation(cmd.cli_ctx)(sdk_no_wait(False, client.begin_create_or_update,
+                                                  resource_group_name, sql_virtual_machine_name, sqlvm_object))
+
+    return client.get(resource_group_name, sql_virtual_machine_name)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sqlvm/custom.py:428]: Will `client.begin_create_create_or_update` return a LROPoller? I think cli core can handle the poller automatically you don't have to do the get yourself.

File: src/azure-cli/azure/cli/command_modules/sqlvm/_validators.py [hunk 3]
```diff
@@ -186,3 +190,322 @@ def validate_assessment_start_time_local(namespace):
...
+    if len(missing_roles) > 0:
+        az_error = InvalidArgumentValueError("The managed identity is lack of the following roles for Azure AD authentication: {}.".format(", ".join(missing_roles)))
+        az_error.set_recommendation("Grant the managed identity EITHER the Directory.Readers role OR the three App roles 'User.Read.All', 'Application.Read.All', 'GroupMember.Read.All'")
+        raise az_error
+
+
+MICROSOFT_GRAPH_API_ERROR = "Unable to validate the permission of MSI due to querying Microsoft Graph API encountered error: {}."
+
+
+def _send(cli_ctx, method, url, param=None, body=None):
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sqlvm/_validators.py:384]: I see a lot of duplicated code here with [Graph Client in role module](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/role/_msgrpah/_graph_client.py)

cc @jiasli to confirm if it's best practice to keep code separately in two modules

Comment (qianliu-pdx) [reply src/azure-cli/azure/cli/command_modules/sqlvm/_validators.py:384]: there was a discussion about this, I have added you to the email thread.

**Review by evelyn-ys — DISMISSED** [review]

**Review by calvinhzy — DISMISSED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**